index.js

```diff
@@ -68,6 +68,10 @@ var validate = module.exports = function(name) {
     warnings.push("name can no longer contain capital letters")
   }
 
+  if (/[~'!()*]/.test(name.split('/').slice(-1)[0])) {
+    warnings.push('name can no longer contain special characters ("~\'!()*")')
```

Comment:
curious about the choice of using `warn` rather than error, I imagine:

1. on the registry we'll block new publications that cause this warning.
2. the CLI will log the message but allow the publish.
3. old packages will continue to be able to be published?

This seems reasonable to me, just refreshing myself with this module.